fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -2986,9 +2986,9 @@ protected boolean planNestedFieldOrComponentChild(@Nonnull KeyExpression child,
             @Nullable ScoredMatch scoredMatch = maybeSortedMatch.apply(null);
             if (scoredMatch != null) {
                 ComparisonRanges nextComparisonRanges = scoredMatch.getComparisonRanges();
-                if (!comparisons.isEqualities() && nextComparisonRanges.getEqualitiesSize() > 0) {
-                    throw new Query.InvalidExpressionException(
-                            "Two nested fields in the same and clause, combine them into one");
+                if (comparisons.getUncommittedComparisonRanges().stream().anyMatch(range -> range.getRangeType() != ComparisonRange.Type.EQUALITY)
```

Comment:
maybe make an `isEqualities(form, to)` instead of using this stream expression twice (here and in `ComparisonRanges`)

Reply:
done